cumulus/ansible/tasks/cluster.py

```diff
@@ -155,22 +155,25 @@ def start_cluster(launch_playbook, provision_playbook, cluster, profile,
         aws_secret_access_key=secret_key,
         cluster_id=cluster['_id'])
 
-    master = parse('head.hosts[0]').find(inventory)
-    if master:
-        master = master[0].value
-    else:
-        raise Exception('Unable to extract cluster master host.')
-
-    status_url = '%s/clusters/%s' % (cumulus.config.girder.baseUrl,
-                                     cluster['_id'])
-    updates = {
-        'config': {
-            'host': master
+    # If master_name has been provide extract out the master host and
+    # update the cluster with it.
+    if master_name:
+        master = parse('%s.hosts[0]' % master_name).find(inventory)
+        if master:
+            master = master[0].value
+        else:
+            raise Exception('Unable to extract cluster master host.')
+
+        status_url = '%s/clusters/%s' % (cumulus.config.girder.baseUrl,
+                                         cluster['_id'])
+        updates = {
+            'config': {
+                'host': master
+            }
```

Comment:
Does it make sense to extend this to include a ```job``` section that contains information about what host to submit jobs too rather than just patching with a generic 'host' field?